docs_nnx/nnx_basics.ipynb

```diff
@@ -657,10 +657,10 @@
     "\n",
     "For example:\n",
     "\n",
-    "- Not every model state can or should be differentiated when interacting with [`jax.grad`](https://jax.readthedocs.io/en/latest/_autosummary/jax.grad.html#jax.grad).\n",
-    "- Or, sometimes, there is a need to specify what part of the model's state is a carry and what part is not when using [`jax.lax.scan`](https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.scan.html#jax.lax.scan).\n",
+    "- Not every model state can or should be differentiated when interacting with `jax.grad`.\n",
+    "- Or, sometimes, there is a need to specify what part of the model's state is a carry and what part is not when using `jax.lax.scan`.\n",
     "\n",
-    "To address this, the Flax NNX API has [`nnx.split`](https://flax.readthedocs.io/en/latest/api_reference/flax.nnx/graph.html#flax.nnx.split), which allows you to pass one or more [`nnx.filterlib.Filter`s](https://flax.readthedocs.io/en/latest/guides/filters_guide.html) to partition the [`nnx.Variable`](https://flax.readthedocs.io/en/latest/api_reference/flax.nnx/variables.html#flax.nnx.Variable)s into mutually exclusive [`nnx.State`](https://flax.readthedocs.io/en/latest/api_reference/flax.nnx/state.html#flax.nnx.State)s. Flax NNX uses `Filter` create `State` groups in APIs (such as [`nnx.split`](https://flax.readthedocs.io/en/latest/api_reference/flax.nnx/graph.html#flax.nnx.split), [`nnx.state()`](https://flax.readthedocs.io/en/latest/api_reference/flax.nnx/graph.html#flax.nnx.state), and many of NNX transforms).\n",
+    "To address this, the Flax NNX API has `nnx.split`, which allows you to pass one or more `Filter`s to partition the `Variable`s into mutually exclusive `State`s. Flax NNx uses `Filter` create `State` groups in APIs (such as `nnx.split`, `nnx.state`, and many of NNX transforms).\n",
```

Comment:
If we don't mention that `Filter` is a Flax NNX `filterlib` `Filter` (i.e. `flax.nnx.filterlib.Filter`), users may find it very difficult to find the reference `Filter` API in our codebase or API docs.

In addition, "`Filter`" (like "`State`", or "`split`") is a common English word, and users may try to do a search and filter for "`Filter`" ... We should specify it's a Flax NNX `filterlib` `Filter`. Naming ia hard. cc @jburnim @jakevdp @IvyZX